File: src/main/java/org/artembogomolova/demo/webapp/config/CustomHibernateValidatorConfiguration.java
package org.artembogomolova.demo.webapp.config;

import javax.validation.Validator;
import org.hibernate.FlushMode;
import org.springframework.boot.autoconfigure.orm.jpa.HibernatePropertiesCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Lazy;

@Configuration
@Lazy
public class CustomHibernateValidatorConfiguration {

  @Bean
  @Lazy
  public HibernatePropertiesCustomizer hibernatePropertiesCustomizer(final Validator validator) {
    return hibernateProperties -> {
      hibernateProperties.put("javax.persistence.validation.factory", validator);
      /*flush only on committed transaction.*/
      hibernateProperties.put("org.hibernate.flushMode", FlushMode.COMMIT.name());
    };
  }
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/config/db/CustomHibernateValidatorConfiguration.kt
package org.artembogomolova.demo.webapp.main.config.db

import org.hibernate.FlushMode
import org.springframework.boot.autoconfigure.orm.jpa.HibernatePropertiesCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Lazy
import javax.validation.Validator

@Configuration
@Lazy
class CustomHibernateValidatorConfiguration {
    @Bean
    @Lazy
    fun hibernatePropertiesCustomizer(validator: Validator?): HibernatePropertiesCustomizer {
        return HibernatePropertiesCustomizer { hibernateProperties: MutableMap<String?, Any?> ->
            hibernateProperties["javax.persistence.validation.factory"] = validator
            /*flush only on committed transaction.*/hibernateProperties["org.hibernate.flushMode"] = FlushMode.COMMIT.name
        }
    }
}